lib/set.sh

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env bash
+
+#####################################################################
+##
+## title: Set Extension
+##
+## description:
+## Set extension of shell (bash)
+##   with well-known function for set manipulation
+##
+## Preconditions:
+##   bash 4.3+
+##
+## author: Mihaly Csokas
+##
+## date: 03. Feb. 2018
+##
+## license: MIT
+##
+#####################################################################
+
+source "./console.sh"
+
+set_structure() {
+  local operation="$1"
+  shift
+
+  case "$operation" in
+
+        create)
+            # pre-conditions:
+            [[ "$#" -lt 1 ]] && log_failure "[create must be followed by one param]" && return 1
+
+            set_name="$1"
+            declare -gA "$set_name"
+
+            ;;
+
+        add)
+            # pre-conditions:
+            [[ "$#" -lt 2 ]] && log_failure "[add must be followed by two params]" && return 1
+
+            declare -n set_name="$1"
+            local key="$2"
+            local value=" "
+            set_name+=( ["$key"]="$value" )
+
+            ;;
+
+        clear)
+            # pre-conditions:
+            [[ "$#" -lt 1 ]] && log_failure "[clear must be followed by one param]" && return 1
+
+            unset $1
+            declare -gA "$1"
```

Comment:
Doksiba rakjuk be, hogy azert hasznalom az `unset` parancsot, hogy torolje a parametert majd azt ujradeklaralom.